src/Components/LoginForm.tsx

```diff
@@ -0,0 +1,233 @@
+
+const addPrevUser = (prevEmail: string) => {
+  const getPrevUsers: string[] | 0 = JSON
+    .parse(localStorage.getItem('prevusers') || '0');
+
+  if (getPrevUsers !== 0) {
+    if (getPrevUsers.length === 5) {
+      getPrevUsers.pop();
+    }
+
+    if (getPrevUsers.filter(mail => mail === prevEmail).length === 0) {
+      getPrevUsers.unshift(prevEmail);
+    }
+
+    localStorage.setItem('prevusers', JSON.stringify(getPrevUsers));
+  } else {
+    localStorage.setItem('prevusers', JSON.stringify([prevEmail]));
+  }
+};
+
+export const LoginForm: React.FC = () => {
+  const { setUser } = useContext(AppContext);
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [newUser, setNewUser] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [emailDisabled, setEmailDisabled] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showPrevUsers, setShowPrevUsers] = useState(false);
+  const hasErrorFromServer = !!errorMessage;
+
+  const ref = useRef<HTMLInputElement>(null);
+
+  let tempEmail = '';
+
+  const prevUsers: string[] | 0 = JSON
+    .parse(localStorage.getItem('prevusers') || '0');
+
+  const toshowPrevUsers = useMemo(() => {
+    return prevUsers !== 0 ? prevUsers
+      .filter(preMail => {
+        return email !== '' ? preMail.includes(email) : true;
+      }) : 0;
+  }, [email]);
+
+  const clearNotification = () => {
+    setErrorMessage('');
+  };
+
+  const newUserObject = {
+    name,
+    username: '',
+    email,
+    phone: '',
+    website: '',
+  };
+
+  const addUser = async (e: FormEvent) => {
+    e.preventDefault();
+    setIsLoading(true);
+
+    const exitingUserLogin = async (emailForLogin: string) => {
+      try {
+        const userExits = await findUser(emailForLogin);
+
+        if (userExits.length !== 0) {
+          localStorage.setItem(
+            'user',
+            JSON.stringify(userExits[0]),
+          );
+          setUser(userExits[0]);
+          addPrevUser(emailForLogin);
+        } else {
+          setNewUser(true);
+          setEmailDisabled(true);
+          setIsLoading(false);
+        }
+      } catch {
+        setErrorMessage('Unable to login');
+        setIsLoading(false);
+      }
+    };
+
+    if (email && !newUser) {
+      exitingUserLogin(email);
+    }
+
+    if (name && newUser) {
+      try {
+        const addedUser = await addUsers(newUserObject);
+
+        if (addedUser) {
+          exitingUserLogin(email);
+        }
+      } catch {
+        setErrorMessage('Unable to login');
+        setIsLoading(false);
+      }
+    }
+  };
+
+  const selectPrevUsers = (prevmail: string) => {
+    if (ref.current) {
+      ref.current.focus();
+    }
+
+    setEmail(prevmail);
+    setShowPrevUsers(false);
+    tempEmail = prevmail;
+  };
+
+  return (
+    <>
+      <form
+        className="box is-size-3 mt-5"
+        onSubmit={(e) => addUser(e)}
```

Comment:
The event will be passed as an argument by default. So you can shorten it
```suggestion
        onSubmit={addUser}
```